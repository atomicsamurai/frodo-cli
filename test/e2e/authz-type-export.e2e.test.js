/**
 * Follow this process to write e2e tests for the CLI project:
 *
 * 1. Test if all the necessary mocks for your tests already exist.
 *    In mock mode, run the command you want to test with the same arguments
 *    and parameters exactly as you want to test it, for example:
 *
 *    $ FRODO_MOCK=1 frodo conn save https://openam-frodo-dev.forgeblocks.com/am volker.scheuber@forgerock.com Sup3rS3cr3t!
 *
 *    If your command completes without errors and with the expected results,
 *    all the required mocks already exist and you are good to write your
 *    test and skip to step #4.
 *
 *    If, however, your command fails and you see errors like the one below,
 *    you know you need to record the mock responses first:
 *
 *    [Polly] [adapter:node-http] Recording for the following request is not found and `recordIfMissing` is `false`.
 *
 * 2. Record mock responses for your exact command.
 *    In mock record mode, run the command you want to test with the same arguments
 *    and parameters exactly as you want to test it, for example:
 *
 *    $ FRODO_MOCK=record frodo conn save https://openam-frodo-dev.forgeblocks.com/am volker.scheuber@forgerock.com Sup3rS3cr3t!
 *
 *    Wait until you see all the Polly instances (mock recording adapters) have
 *    shutdown before you try to run step #1 again.
 *    Messages like these indicate mock recording adapters shutting down:
 *
 *    Polly instance 'conn/4' stopping in 3s...
 *    Polly instance 'conn/4' stopping in 2s...
 *    Polly instance 'conn/save/3' stopping in 3s...
 *    Polly instance 'conn/4' stopping in 1s...
 *    Polly instance 'conn/save/3' stopping in 2s...
 *    Polly instance 'conn/4' stopped.
 *    Polly instance 'conn/save/3' stopping in 1s...
 *    Polly instance 'conn/save/3' stopped.
 *
 * 3. Validate your freshly recorded mock responses are complete and working.
 *    Re-run the exact command you want to test in mock mode (see step #1).
 *
 * 4. Write your test.
 *    Make sure to use the exact command including number of arguments and params.
 *
 * 5. Commit both your test and your new recordings to the repository.
 *    Your tests are likely going to reside outside the frodo-lib project but
 *    the recordings must be committed to the frodo-lib project.
 */

/*
FRODO_MOCK=record FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am FRODO_SA_ID=b672336b-41ef-428d-ae4a-e0c082875377 FRODO_SA_JWK=$(<~/Downloads/frodo-test_privateKey.jwk) frodo authz type export -n URL
FRODO_MOCK=record FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am FRODO_SA_ID=b672336b-41ef-428d-ae4a-e0c082875377 FRODO_SA_JWK=$(<~/Downloads/frodo-test_privateKey.jwk) frodo authz type export --type-name URL
FRODO_MOCK=record FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am FRODO_SA_ID=b672336b-41ef-428d-ae4a-e0c082875377 FRODO_SA_JWK=$(<~/Downloads/frodo-test_privateKey.jwk) frodo authz type export -n URL -f test.json
FRODO_MOCK=record FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am FRODO_SA_ID=b672336b-41ef-428d-ae4a-e0c082875377 FRODO_SA_JWK=$(<~/Downloads/frodo-test_privateKey.jwk) frodo authz type export -i 76656a38-5f8e-401b-83aa-4ccb74ce88d2
FRODO_MOCK=record FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am FRODO_SA_ID=b672336b-41ef-428d-ae4a-e0c082875377 FRODO_SA_JWK=$(<~/Downloads/frodo-test_privateKey.jwk) frodo authz type export --type-id 76656a38-5f8e-401b-83aa-4ccb74ce88d2
FRODO_MOCK=record FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am FRODO_SA_ID=b672336b-41ef-428d-ae4a-e0c082875377 FRODO_SA_JWK=$(<~/Downloads/frodo-test_privateKey.jwk) frodo authz type export -i 76656a38-5f8e-401b-83aa-4ccb74ce88d2 -f test.json
FRODO_MOCK=record FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am FRODO_SA_ID=b672336b-41ef-428d-ae4a-e0c082875377 FRODO_SA_JWK=$(<~/Downloads/frodo-test_privateKey.jwk) frodo authz type export -a
FRODO_MOCK=record FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am FRODO_SA_ID=b672336b-41ef-428d-ae4a-e0c082875377 FRODO_SA_JWK=$(<~/Downloads/frodo-test_privateKey.jwk) frodo authz type export --all
FRODO_MOCK=record FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am FRODO_SA_ID=b672336b-41ef-428d-ae4a-e0c082875377 FRODO_SA_JWK=$(<~/Downloads/frodo-test_privateKey.jwk) frodo authz type export -a --file test.json
FRODO_MOCK=record FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am FRODO_SA_ID=b672336b-41ef-428d-ae4a-e0c082875377 FRODO_SA_JWK=$(<~/Downloads/frodo-test_privateKey.jwk) frodo authz type export -A
FRODO_MOCK=record FRODO_HOST=https://openam-frodo-dev.forgeblocks.com/am FRODO_SA_ID=b672336b-41ef-428d-ae4a-e0c082875377 FRODO_SA_JWK=$(<~/Downloads/frodo-test_privateKey.jwk) frodo authz type export --all-separate
*/
import { testExport } from './utils/TestUtils';
import { connection as c } from './utils/TestConfig';

process.env['FRODO_MOCK'] = '1';
const env = {
    env: process.env,
};
env.env.FRODO_HOST = c.host;
env.env.FRODO_SA_ID = c.saId;
env.env.FRODO_SA_JWK = c.saJwk;

const type = 'resourcetype.authz';

describe('frodo authz type export', () => {
    test('"frodo authz type export -n URL": should export the resource type named "URL"', async () => {
        const CMD = `frodo authz type export -n URL`;
        const exportFile = "URL.resourcetype.authz.json";
        await testExport(CMD, env, type, exportFile);
    });

    test('"frodo authz type export --type-name URL": should export the resource type named "URL"', async () => {
        const CMD = `frodo authz type export --type-name URL`;
        const exportFile = "URL.resourcetype.authz.json";
        await testExport(CMD, env, type, exportFile);
    });

    test('"frodo authz type export -n URL -f test.json": should export the resource type named "URL" into file named test.json', async () => {
        const CMD = `frodo authz type export -n URL -f test.json`;
        const exportFile = "test.json";
        await testExport(CMD, env, type, exportFile);
    });

    test('"frodo authz type export -i 76656a38-5f8e-401b-83aa-4ccb74ce88d2": should export the resource type with id "76656a38-5f8e-401b-83aa-4ccb74ce88d2"', async () => {
        const CMD = `frodo authz type export -i 76656a38-5f8e-401b-83aa-4ccb74ce88d2`;
        const exportFile = "76656a38-5f8e-401b-83aa-4ccb74ce88d2.resourcetype.authz.json";
        await testExport(CMD, env, type, exportFile);
    });

    test('"frodo authz type export --type-id 76656a38-5f8e-401b-83aa-4ccb74ce88d2": should export the resource type with id "76656a38-5f8e-401b-83aa-4ccb74ce88d2"', async () => {
        const CMD = `frodo authz type export --type-id 76656a38-5f8e-401b-83aa-4ccb74ce88d2`;
        const exportFile = "76656a38-5f8e-401b-83aa-4ccb74ce88d2.resourcetype.authz.json";
        await testExport(CMD, env, type, exportFile);
    });

    test('"frodo authz type export -i 76656a38-5f8e-401b-83aa-4ccb74ce88d2 -f test.json": should export the resource type with id "76656a38-5f8e-401b-83aa-4ccb74ce88d2" into file named test.json', async () => {
        const CMD = `frodo authz type export -i 76656a38-5f8e-401b-83aa-4ccb74ce88d2 -f test.json`;
        const exportFile = "test.json";
        await testExport(CMD, env, type, exportFile);
    });

    test('"frodo authz type export -a": should export all resource types to a single file', async () => {
        const CMD = `frodo authz type export -a`;
        const exportFile = "allAlphaResourceTypes.resourcetype.authz.json";
        await testExport(CMD, env, type, exportFile);
    });

    test('"frodo authz type export --all": should export all resource types to a single file', async () => {
        const CMD = `frodo authz type export --all`;
        const exportFile = "allAlphaResourceTypes.resourcetype.authz.json";
        await testExport(CMD, env, type, exportFile);
    });

    test('"frodo authz type export -a --file test.json": should export all resource types to a single file named test.json', async () => {
        const CMD = `frodo authz type export -a --file test.json`;
        const exportFile = "test.json";
        await testExport(CMD, env, type, exportFile);
    });

    test('"frodo authz type export -A": should export all resource types to separate files', async () => {
        const CMD = `frodo authz type export -A`;
        await testExport(CMD, env, type);
    });

    test('"frodo authz type export --all-separate": should export all resource types to separate files', async () => {
        const CMD = `frodo authz type export --all-separate`;
        await testExport(CMD, env, type);
    });
});
