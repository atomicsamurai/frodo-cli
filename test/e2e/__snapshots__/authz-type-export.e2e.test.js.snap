// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`frodo authz type export "frodo authz type export --all": should export all resource types to a single file 1`] = `""`;

exports[`frodo authz type export "frodo authz type export --all": should export all resource types to a single file 2`] = `
{
  "meta": Any<Object>,
  "policy": {},
  "policyset": {},
  "resourcetype": {
    "0aa5ed25-0c62-4ff5-9a42-3bda8c5cbb76": {
      "actions": {
        "action1": true,
        "action2": true,
      },
      "createdBy": "id=8d9723a9-a439-4cbf-beb4-30e52811789d,ou=user,ou=am-config",
      "creationDate": 1684544147346,
      "description": "Frodo Test Resource Type Thirteen",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717616300,
      "name": "FrodoTestResourceType13",
      "patterns": [
        "pattern2://*:*/*?*",
        "pattern1://*:*/*",
      ],
      "uuid": "0aa5ed25-0c62-4ff5-9a42-3bda8c5cbb76",
    },
    "119b291c-40b3-4b1e-8d84-c2a561a2cb1f": {
      "actions": {
        "action1": true,
        "action2": true,
      },
      "createdBy": "id=8d9723a9-a439-4cbf-beb4-30e52811789d,ou=user,ou=am-config",
      "creationDate": 1684544147437,
      "description": "Frodo Test Resource Type Fourteen",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717616482,
      "name": "FrodoTestResourceType14",
      "patterns": [
        "pattern2://*:*/*?*",
        "pattern1://*:*/*",
      ],
      "uuid": "119b291c-40b3-4b1e-8d84-c2a561a2cb1f",
    },
    "3c5f13af-ca17-403e-b47d-d15263cce954": {
      "actions": {
        "action1": true,
        "action2": true,
      },
      "createdBy": "id=8d9723a9-a439-4cbf-beb4-30e52811789d,ou=user,ou=am-config",
      "creationDate": 1684545571229,
      "description": "Frodo Test Resource Type Fifteen",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717616660,
      "name": "FrodoTestResourceType15",
      "patterns": [
        "pattern2://*:*/*?*",
        "pattern1://*:*/*",
      ],
      "uuid": "3c5f13af-ca17-403e-b47d-d15263cce954",
    },
    "3fc799d7-b73f-49e0-a70b-e37990e54e56": {
      "actions": {
        "action1": true,
        "action2": true,
      },
      "createdBy": "id=8d9723a9-a439-4cbf-beb4-30e52811789d,ou=user,ou=am-config",
      "creationDate": 1684544147204,
      "description": "Frodo Test Resource Type Twelve",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717616120,
      "name": "FrodoTestResourceType12",
      "patterns": [
        "pattern2://*:*/*?*",
        "pattern1://*:*/*",
      ],
      "uuid": "3fc799d7-b73f-49e0-a70b-e37990e54e56",
    },
    "76656a38-5f8e-401b-83aa-4ccb74ce88d2": {
      "actions": {
        "DELETE": true,
        "GET": true,
        "HEAD": true,
        "OPTIONS": true,
        "PATCH": true,
        "POST": true,
        "PUT": true,
      },
      "createdBy": "id=dsameuser,ou=user,ou=am-config",
      "creationDate": 1595479030487,
      "description": "The built-in URL Resource Type available to OpenAMPolicies.",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717661706,
      "name": "URL",
      "patterns": [
        "*://*:*/*",
        "*://*:*/*?*",
      ],
      "uuid": "76656a38-5f8e-401b-83aa-4ccb74ce88d2",
    },
    "993eba78-1c3f-4f27-b205-b4b29418f831": {
      "actions": {
        "action1": true,
        "action2": true,
      },
      "createdBy": "id=8d9723a9-a439-4cbf-beb4-30e52811789d,ou=user,ou=am-config",
      "creationDate": 1684544147077,
      "description": "Frodo Test Resource Type Eleven",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717615932,
      "name": "FrodoTestResourceType11",
      "patterns": [
        "pattern2://*:*/*?*",
        "pattern1://*:*/*",
      ],
      "uuid": "993eba78-1c3f-4f27-b205-b4b29418f831",
    },
    "d60b7a71-1dc6-44a5-8e48-e4b9d92dee8b": {
      "actions": {
        "GRANT": true,
      },
      "createdBy": "id=dsameuser,ou=user,ou=am-config",
      "creationDate": 1595479030586,
      "description": "The built-in OAuth2 Scope Resource Type for OAuth2policy-provided scope.",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717661536,
      "name": "OAuth2 Scope",
      "patterns": [
        "*://*:*/*",
        "*://*:*/*?*",
        "*",
      ],
      "uuid": "d60b7a71-1dc6-44a5-8e48-e4b9d92dee8b",
    },
  },
  "script": {},
}
`;

exports[`frodo authz type export "frodo authz type export --type-id 76656a38-5f8e-401b-83aa-4ccb74ce88d2": should export the resource type with id "76656a38-5f8e-401b-83aa-4ccb74ce88d2" 1`] = `""`;

exports[`frodo authz type export "frodo authz type export --type-id 76656a38-5f8e-401b-83aa-4ccb74ce88d2": should export the resource type with id "76656a38-5f8e-401b-83aa-4ccb74ce88d2" 2`] = `
{
  "meta": Any<Object>,
  "policy": {},
  "policyset": {},
  "resourcetype": {
    "76656a38-5f8e-401b-83aa-4ccb74ce88d2": {
      "actions": {
        "DELETE": true,
        "GET": true,
        "HEAD": true,
        "OPTIONS": true,
        "PATCH": true,
        "POST": true,
        "PUT": true,
      },
      "createdBy": "id=dsameuser,ou=user,ou=am-config",
      "creationDate": 1595479030487,
      "description": "The built-in URL Resource Type available to OpenAMPolicies.",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717661706,
      "name": "URL",
      "patterns": [
        "*://*:*/*",
        "*://*:*/*?*",
      ],
      "uuid": "76656a38-5f8e-401b-83aa-4ccb74ce88d2",
    },
  },
  "script": {},
}
`;

exports[`frodo authz type export "frodo authz type export --type-name URL -f test.json": should export the resource type named "URL" 1`] = `""`;

exports[`frodo authz type export "frodo authz type export --type-name URL -f test.json": should export the resource type named "URL" 2`] = `
{
  "meta": Any<Object>,
  "policy": {},
  "policyset": {},
  "resourcetype": {
    "76656a38-5f8e-401b-83aa-4ccb74ce88d2": {
      "actions": {
        "DELETE": true,
        "GET": true,
        "HEAD": true,
        "OPTIONS": true,
        "PATCH": true,
        "POST": true,
        "PUT": true,
      },
      "createdBy": "id=dsameuser,ou=user,ou=am-config",
      "creationDate": 1595479030487,
      "description": "The built-in URL Resource Type available to OpenAMPolicies.",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717661706,
      "name": "URL",
      "patterns": [
        "*://*:*/*",
        "*://*:*/*?*",
      ],
      "uuid": "76656a38-5f8e-401b-83aa-4ccb74ce88d2",
    },
  },
  "script": {},
}
`;

exports[`frodo authz type export "frodo authz type export -A": should export all resource types to separate files 1`] = `""`;

exports[`frodo authz type export "frodo authz type export -A": should export all resource types to separate files 2`] = `
{
  "meta": Any<Object>,
  "policy": {},
  "policyset": {},
  "resourcetype": {
    "993eba78-1c3f-4f27-b205-b4b29418f831": {
      "actions": {
        "action1": true,
        "action2": true,
      },
      "createdBy": "id=8d9723a9-a439-4cbf-beb4-30e52811789d,ou=user,ou=am-config",
      "creationDate": 1684544147077,
      "description": "Frodo Test Resource Type Eleven",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717615932,
      "name": "FrodoTestResourceType11",
      "patterns": [
        "pattern2://*:*/*?*",
        "pattern1://*:*/*",
      ],
      "uuid": "993eba78-1c3f-4f27-b205-b4b29418f831",
    },
  },
  "script": {},
}
`;

exports[`frodo authz type export "frodo authz type export -A": should export all resource types to separate files 3`] = `
{
  "meta": Any<Object>,
  "policy": {},
  "policyset": {},
  "resourcetype": {
    "3fc799d7-b73f-49e0-a70b-e37990e54e56": {
      "actions": {
        "action1": true,
        "action2": true,
      },
      "createdBy": "id=8d9723a9-a439-4cbf-beb4-30e52811789d,ou=user,ou=am-config",
      "creationDate": 1684544147204,
      "description": "Frodo Test Resource Type Twelve",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717616120,
      "name": "FrodoTestResourceType12",
      "patterns": [
        "pattern2://*:*/*?*",
        "pattern1://*:*/*",
      ],
      "uuid": "3fc799d7-b73f-49e0-a70b-e37990e54e56",
    },
  },
  "script": {},
}
`;

exports[`frodo authz type export "frodo authz type export -A": should export all resource types to separate files 4`] = `
{
  "meta": Any<Object>,
  "policy": {},
  "policyset": {},
  "resourcetype": {
    "0aa5ed25-0c62-4ff5-9a42-3bda8c5cbb76": {
      "actions": {
        "action1": true,
        "action2": true,
      },
      "createdBy": "id=8d9723a9-a439-4cbf-beb4-30e52811789d,ou=user,ou=am-config",
      "creationDate": 1684544147346,
      "description": "Frodo Test Resource Type Thirteen",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717616300,
      "name": "FrodoTestResourceType13",
      "patterns": [
        "pattern2://*:*/*?*",
        "pattern1://*:*/*",
      ],
      "uuid": "0aa5ed25-0c62-4ff5-9a42-3bda8c5cbb76",
    },
  },
  "script": {},
}
`;

exports[`frodo authz type export "frodo authz type export -A": should export all resource types to separate files 5`] = `
{
  "meta": Any<Object>,
  "policy": {},
  "policyset": {},
  "resourcetype": {
    "119b291c-40b3-4b1e-8d84-c2a561a2cb1f": {
      "actions": {
        "action1": true,
        "action2": true,
      },
      "createdBy": "id=8d9723a9-a439-4cbf-beb4-30e52811789d,ou=user,ou=am-config",
      "creationDate": 1684544147437,
      "description": "Frodo Test Resource Type Fourteen",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717616482,
      "name": "FrodoTestResourceType14",
      "patterns": [
        "pattern2://*:*/*?*",
        "pattern1://*:*/*",
      ],
      "uuid": "119b291c-40b3-4b1e-8d84-c2a561a2cb1f",
    },
  },
  "script": {},
}
`;

exports[`frodo authz type export "frodo authz type export -A": should export all resource types to separate files 6`] = `
{
  "meta": Any<Object>,
  "policy": {},
  "policyset": {},
  "resourcetype": {
    "3c5f13af-ca17-403e-b47d-d15263cce954": {
      "actions": {
        "action1": true,
        "action2": true,
      },
      "createdBy": "id=8d9723a9-a439-4cbf-beb4-30e52811789d,ou=user,ou=am-config",
      "creationDate": 1684545571229,
      "description": "Frodo Test Resource Type Fifteen",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717616660,
      "name": "FrodoTestResourceType15",
      "patterns": [
        "pattern2://*:*/*?*",
        "pattern1://*:*/*",
      ],
      "uuid": "3c5f13af-ca17-403e-b47d-d15263cce954",
    },
  },
  "script": {},
}
`;

exports[`frodo authz type export "frodo authz type export -A": should export all resource types to separate files 7`] = `
{
  "meta": Any<Object>,
  "policy": {},
  "policyset": {},
  "resourcetype": {
    "d60b7a71-1dc6-44a5-8e48-e4b9d92dee8b": {
      "actions": {
        "GRANT": true,
      },
      "createdBy": "id=dsameuser,ou=user,ou=am-config",
      "creationDate": 1595479030586,
      "description": "The built-in OAuth2 Scope Resource Type for OAuth2policy-provided scope.",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717661536,
      "name": "OAuth2 Scope",
      "patterns": [
        "*://*:*/*",
        "*://*:*/*?*",
        "*",
      ],
      "uuid": "d60b7a71-1dc6-44a5-8e48-e4b9d92dee8b",
    },
  },
  "script": {},
}
`;

exports[`frodo authz type export "frodo authz type export -A": should export all resource types to separate files 8`] = `
{
  "meta": Any<Object>,
  "policy": {},
  "policyset": {},
  "resourcetype": {
    "76656a38-5f8e-401b-83aa-4ccb74ce88d2": {
      "actions": {
        "DELETE": true,
        "GET": true,
        "HEAD": true,
        "OPTIONS": true,
        "PATCH": true,
        "POST": true,
        "PUT": true,
      },
      "createdBy": "id=dsameuser,ou=user,ou=am-config",
      "creationDate": 1595479030487,
      "description": "The built-in URL Resource Type available to OpenAMPolicies.",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717661706,
      "name": "URL",
      "patterns": [
        "*://*:*/*",
        "*://*:*/*?*",
      ],
      "uuid": "76656a38-5f8e-401b-83aa-4ccb74ce88d2",
    },
  },
  "script": {},
}
`;

exports[`frodo authz type export "frodo authz type export -a --file my-allAlphaResourceTypes.resourcetype.authz.json": should export all resource types to a single file named my-allAlphaResourceTypes.resourcetype.authz.json 1`] = `""`;

exports[`frodo authz type export "frodo authz type export -a --file my-allAlphaResourceTypes.resourcetype.authz.json": should export all resource types to a single file named my-allAlphaResourceTypes.resourcetype.authz.json 2`] = `
{
  "meta": Any<Object>,
  "policy": {},
  "policyset": {},
  "resourcetype": {
    "0aa5ed25-0c62-4ff5-9a42-3bda8c5cbb76": {
      "actions": {
        "action1": true,
        "action2": true,
      },
      "createdBy": "id=8d9723a9-a439-4cbf-beb4-30e52811789d,ou=user,ou=am-config",
      "creationDate": 1684544147346,
      "description": "Frodo Test Resource Type Thirteen",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717616300,
      "name": "FrodoTestResourceType13",
      "patterns": [
        "pattern2://*:*/*?*",
        "pattern1://*:*/*",
      ],
      "uuid": "0aa5ed25-0c62-4ff5-9a42-3bda8c5cbb76",
    },
    "119b291c-40b3-4b1e-8d84-c2a561a2cb1f": {
      "actions": {
        "action1": true,
        "action2": true,
      },
      "createdBy": "id=8d9723a9-a439-4cbf-beb4-30e52811789d,ou=user,ou=am-config",
      "creationDate": 1684544147437,
      "description": "Frodo Test Resource Type Fourteen",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717616482,
      "name": "FrodoTestResourceType14",
      "patterns": [
        "pattern2://*:*/*?*",
        "pattern1://*:*/*",
      ],
      "uuid": "119b291c-40b3-4b1e-8d84-c2a561a2cb1f",
    },
    "3c5f13af-ca17-403e-b47d-d15263cce954": {
      "actions": {
        "action1": true,
        "action2": true,
      },
      "createdBy": "id=8d9723a9-a439-4cbf-beb4-30e52811789d,ou=user,ou=am-config",
      "creationDate": 1684545571229,
      "description": "Frodo Test Resource Type Fifteen",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717616660,
      "name": "FrodoTestResourceType15",
      "patterns": [
        "pattern2://*:*/*?*",
        "pattern1://*:*/*",
      ],
      "uuid": "3c5f13af-ca17-403e-b47d-d15263cce954",
    },
    "3fc799d7-b73f-49e0-a70b-e37990e54e56": {
      "actions": {
        "action1": true,
        "action2": true,
      },
      "createdBy": "id=8d9723a9-a439-4cbf-beb4-30e52811789d,ou=user,ou=am-config",
      "creationDate": 1684544147204,
      "description": "Frodo Test Resource Type Twelve",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717616120,
      "name": "FrodoTestResourceType12",
      "patterns": [
        "pattern2://*:*/*?*",
        "pattern1://*:*/*",
      ],
      "uuid": "3fc799d7-b73f-49e0-a70b-e37990e54e56",
    },
    "76656a38-5f8e-401b-83aa-4ccb74ce88d2": {
      "actions": {
        "DELETE": true,
        "GET": true,
        "HEAD": true,
        "OPTIONS": true,
        "PATCH": true,
        "POST": true,
        "PUT": true,
      },
      "createdBy": "id=dsameuser,ou=user,ou=am-config",
      "creationDate": 1595479030487,
      "description": "The built-in URL Resource Type available to OpenAMPolicies.",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717661706,
      "name": "URL",
      "patterns": [
        "*://*:*/*",
        "*://*:*/*?*",
      ],
      "uuid": "76656a38-5f8e-401b-83aa-4ccb74ce88d2",
    },
    "993eba78-1c3f-4f27-b205-b4b29418f831": {
      "actions": {
        "action1": true,
        "action2": true,
      },
      "createdBy": "id=8d9723a9-a439-4cbf-beb4-30e52811789d,ou=user,ou=am-config",
      "creationDate": 1684544147077,
      "description": "Frodo Test Resource Type Eleven",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717615932,
      "name": "FrodoTestResourceType11",
      "patterns": [
        "pattern2://*:*/*?*",
        "pattern1://*:*/*",
      ],
      "uuid": "993eba78-1c3f-4f27-b205-b4b29418f831",
    },
    "d60b7a71-1dc6-44a5-8e48-e4b9d92dee8b": {
      "actions": {
        "GRANT": true,
      },
      "createdBy": "id=dsameuser,ou=user,ou=am-config",
      "creationDate": 1595479030586,
      "description": "The built-in OAuth2 Scope Resource Type for OAuth2policy-provided scope.",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717661536,
      "name": "OAuth2 Scope",
      "patterns": [
        "*://*:*/*",
        "*://*:*/*?*",
        "*",
      ],
      "uuid": "d60b7a71-1dc6-44a5-8e48-e4b9d92dee8b",
    },
  },
  "script": {},
}
`;

exports[`frodo authz type export "frodo authz type export -i 76656a38-5f8e-401b-83aa-4ccb74ce88d2 -f my-76656a38-5f8e-401b-83aa-4ccb74ce88d2.resourcetype.authz.json": should export the resource type with id "76656a38-5f8e-401b-83aa-4ccb74ce88d2" into file named my-76656a38-5f8e-401b-83aa-4ccb74ce88d2.resourcetype.authz.json 1`] = `""`;

exports[`frodo authz type export "frodo authz type export -i 76656a38-5f8e-401b-83aa-4ccb74ce88d2 -f my-76656a38-5f8e-401b-83aa-4ccb74ce88d2.resourcetype.authz.json": should export the resource type with id "76656a38-5f8e-401b-83aa-4ccb74ce88d2" into file named my-76656a38-5f8e-401b-83aa-4ccb74ce88d2.resourcetype.authz.json 2`] = `
{
  "meta": Any<Object>,
  "policy": {},
  "policyset": {},
  "resourcetype": {
    "76656a38-5f8e-401b-83aa-4ccb74ce88d2": {
      "actions": {
        "DELETE": true,
        "GET": true,
        "HEAD": true,
        "OPTIONS": true,
        "PATCH": true,
        "POST": true,
        "PUT": true,
      },
      "createdBy": "id=dsameuser,ou=user,ou=am-config",
      "creationDate": 1595479030487,
      "description": "The built-in URL Resource Type available to OpenAMPolicies.",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717661706,
      "name": "URL",
      "patterns": [
        "*://*:*/*",
        "*://*:*/*?*",
      ],
      "uuid": "76656a38-5f8e-401b-83aa-4ccb74ce88d2",
    },
  },
  "script": {},
}
`;

exports[`frodo authz type export "frodo authz type export -n URL": should export the resource type named "URL" 1`] = `""`;

exports[`frodo authz type export "frodo authz type export -n URL": should export the resource type named "URL" 2`] = `
{
  "meta": Any<Object>,
  "policy": {},
  "policyset": {},
  "resourcetype": {
    "76656a38-5f8e-401b-83aa-4ccb74ce88d2": {
      "actions": {
        "DELETE": true,
        "GET": true,
        "HEAD": true,
        "OPTIONS": true,
        "PATCH": true,
        "POST": true,
        "PUT": true,
      },
      "createdBy": "id=dsameuser,ou=user,ou=am-config",
      "creationDate": 1595479030487,
      "description": "The built-in URL Resource Type available to OpenAMPolicies.",
      "lastModifiedBy": "id=b672336b-41ef-428d-ae4a-e0c082875377,ou=user,ou=am-config",
      "lastModifiedDate": 1694717661706,
      "name": "URL",
      "patterns": [
        "*://*:*/*",
        "*://*:*/*?*",
      ],
      "uuid": "76656a38-5f8e-401b-83aa-4ccb74ce88d2",
    },
  },
  "script": {},
}
`;
